### ISSUE TITLE
[Feature]: Behaviour of "Shuffle" button

### ISSUE THREAD

**Issue by KOLIWAN** [fast4x/RiMusic#2867]
### Description

Right now there's three ways to enable Shuffle:

1) At the playlist header there's a button to shuffle. If you press it, the playlist starts shuffled.

![Screenshot_20240722_203911](https://github.com/user-attachments/assets/eb4926e2-3d4d-4023-b479-862baae46a9d)

2) In the player there's a Shuffle button. If you press it it'll shuffle your playlist and start play random song.

![Screenshot_20240722_203920](https://github.com/user-attachments/assets/fd6b38f8-7489-458e-b3c2-89252337876d)

3) Shuffle button in notifications/lock screen. If you press it, it'd shuffle playlist, but the song continue playing. The button changes icon to "enabled". After you push button one more time, button icon becomes "disabled".

![Screenshot_20240722_203930](https://github.com/user-attachments/assets/c7151779-2720-4d9a-bdb0-7238c8d1bb64)

### Suggested Solution

Right now, all three Shuffle buttons work independently of each other, and this is confusing. I suggest bringing them to the same behavioral logic.
What is my idea:

1) Shuffle button in playlist's header is working fine and as expected, you push it -> playlist start playing in random order.

2) The biggest problem with the Shuffle button in the player. Usually in players when you press Shuffle, the playlist changes to random order, but the current song remains playing. I suggest to make the same behavior, so that the Shuffle button can be clicked to enable random song order and unclicked to return to normal order. Now when you press Shuffle in player new song from queue starts playing and order becomes random, there's no opportunity to undo this.

3) Shuffle button in notifications/lock screen can be "enabled" and "disabled", as I expect to see in player. However, if you enable Shuffle somewhere else (in player or playlist header) it isn't showed in notifications mini player, Shuffle button is "disabled". It is expected that if Shuffle is enabled, the button will be “pressed” in notifications/lock screen as well.

Summary: it is expected that when Shuffle is enabled anywhere, it will show up as a “pressed” button in the player and in the notifications/lock screen. When the button is pressed again, it goes to the “disabled” state, the original play queue is restored.

### Alternatives

Alternatives are open for discussion...

**Comment by fast4x** [comment]
Yes you are right, shuffle in notification area it's different because it enable shuffle mode, that shuffle songs before playing next...
